pvlib/tracking.py

```diff
@@ -570,3 +512,152 @@ def singleaxis(apparent_zenith, apparent_azimuth,
         out = {k: np.where(zen_gt_90, np.nan, v) for k, v in out.items()}
 
     return out
+
+
+def calc_axis_tilt(system_azimuth, system_zenith, axis_azimuth):
+    """
+    Calculate tracker axis tilt in the global reference frame when on a sloped
+    plane.
+
+    Parameters
+    ----------
+    system_azimuth : float
+        direction of normal to slope on horizontal [degrees]
```

Comment:
I think `slope_azimuth` and `slope_zenith` would be better. 

suggest "azimuth of the projection to horizontal of the slope normal" The words "azimuth" and "projection" seem needed here, as well as definition of "slope normal" or "normal to slope". An example would help.

Reply:
I'm partial to "system" because IMO it's the PV system that's sloped, and the ground underneath the system might have a different slope, although I believe in practice, they are commonly parallel.

Reply:
I agree with your point, but, I'm not assuming it's ground slope or system slope. `system_azimuth` is too generic IMO, at first glance (before I read all the argument descriptions) I thought this was the tracker axis azimuth.

Now that I think further I'm not clear on what this function's purpose is. It's calculating the angle from horizontal in a particular direction along a plane. But we earlier defined the "system plane" to be the plane containing the tracker axes implying that axis_tilt = 0, so here "system plane" means something different?

Reply:
**I'm okay with "slope" azimuth & zenith instead of "system"** b/c in practice, I believe most if not all systems will be orientated parallel to the slope. Also in the NREL tech report we call this plane the "grade slope" with suffix "g" - so for consistency, perhaps we should use the same terminology, _ie_ "slope". **I'm also okay with "tilt" instead of "zenith".** -- @kanderso-nrel, care to comment?

As far as the purpose of `calc_axis_tilt()` -- `axis_tilt` is a required input for `singleaxis()` but it's not intuitively obvious to me what it should be for a system on a slope that was not parallel to the tracker axis azimuth. Consider the system in the image below:

![image](https://user-images.githubusercontent.com/1385621/92173094-c74f0500-edf0-11ea-9c55-2fcfb5793b6f.png)

(@kanderso-nrel pls correct me if I'm inconsistent, thx)

The `axis_slope` is &beta;<sub>a</sub> while the slope (or system) tilt is &beta;<sub>g</sub>. The `side_slope` is &beta;<sub>c</sub> where the "c" stands for "cross-axis", _ie_ to the "side" perpendicular to the tracker axis. The slope azimuth (CD) is not parallel to the tracker axis azimuth BD, so given only the slope (or system) orientation, we would use `calc_axis_tilt()` to get the required input for `singleaxis`. Also because the tracker rows (thin black lines) are constrained to the slope (the grey plane), the `side_slope` is also required by `singleaxis()` or else the trackers will shade each other as indicated in Kevin's paper: **[Maximizing Yield with Improved Single-Axis
Backtracking on Cross-Axis Slopes](https://www.nrel.gov/docs/fy20osti/76023.pdf)**.

Reply:
> I'm okay with "slope" azimuth & zenith instead of "system"... I'm also okay with "tilt" instead of "zenith"

Slope/tilt are fine with me.  I agree that `system_azimuth` is easily interpreted to mean tracker axis azimuth.

> Now that I think further I'm not clear on what this function's purpose is. It's calculating the angle from horizontal in a particular direction along a plane. But we earlier defined the "system plane" to be the plane containing the tracker axes implying that axis_tilt = 0, so here "system plane" means something different?

I agree with @mikofski's explanation above.  The motivating example in my mind is a N-S axis array on a parcel with a mild slope to the SE.  Given the axis azimuth (180), the slope azimuth (~135), and the steepness of the slope, what is the axis tilt?  This is assuming a mindset where the system must be built with a certain azimuth, but the tilt can do as it pleases to match the terrain.

Reply:
I see now where my confusion arose - we are talking about two different angles for `axis_tilt`. `axis_tilt` is defined as the angle _from horizontal_ of the tracker axis. This the same definition as before. I was picturing `axis_tilt` as the angle between the tracker axis and the ground plane, i.e., the angle formed by elevating one end of the tracker. I'm not at all sure if trackers with an axis_tilt are common in the industry. 

For trackers on a horizontal surface, the "angle between tracker axis and ground" is a valid interpretation of `axis_tilt`. This PR will retain the same definition of `axis_tilt` (angle from horizontal) but its interpretation is not the same as before.

I think this has circled around to the discussion we had about the NREL report, that the system plane is assumed to be parallel to the overall slope of the ground, since there is no longer a parameter that provides an elevation of one end of the tracker axis.

Reply:
> but the tilt can do as it pleases to match the terrain.

Yes, and this is a change in the meaning of `axis_tilt` from the previous `singleaxis` function.

Reply:
The "famous" example of tilted-axis tracking is the array of T20 trackers at Nellis AFB, but I have the impression that this sort of "artificially" tilted axis is very uncommon.  I also have the impression (and I think @mikofski does as well?) that developers are increasingly more willing to develop sloped parcels than they were in years past.

> I think this has circled around to the discussion we had about the NREL report, that the system plane is assumed to be parallel to the overall slope of the ground, since there is no longer a parameter that provides an elevation of one end of the tracker axis.

Small quibble: the system plane is indeed assumed to be parallel to the overall slope of the ground in `calc_tracker_axis_tilt`, but `singleaxis` is only concerned about the axis geometry in the global reference frame.  So if someone wants to model the T20 trackers at Nellis, the new `singleaxis` can still do that -- `singleaxis` doesn't care if `axis_tilt` is from raising up one end, or from installing on sloped ground, or both.  But if someone wants to model the T20 trackers installed on sloped terrain, they will have to calculate the overall `axis_tilt` themselves.